Src/FluentAssertions/Primitives/TimeOnlyAssertions.cs

```diff
@@ -127,6 +127,100 @@ public AndConstraint<TAssertions> Be(TimeOnly? expected, string because = "", pa
         return new AndConstraint<TAssertions>((TAssertions)this);
     }
 
+    /// <summary>
+    /// Asserts that the current <see cref="TimeOnly"/>  is within the specified time
+    /// from the specified <paramref name="nearbyTime"/> value.
+    /// </summary>
+    /// <param name="nearbyTime">
+    /// The expected time to compare the actual value with.
+    /// </param>
+    /// <param name="precision">
+    /// The maximum amount of time which the two values may differ.
+    /// </param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public AndConstraint<TAssertions> BeCloseTo(TimeOnly nearbyTime, TimeSpan precision, string because = "",
+        params object[] becauseArgs)
+    {
+        if (precision < TimeSpan.Zero)
+        {
+            throw new ArgumentOutOfRangeException(nameof(precision), $"The value of {nameof(precision)} must be non-negative.");
+        }
+
+        long distanceToMinInTicks = (nearbyTime - TimeOnly.MinValue).Ticks;
```

Comment:
🤔 I'm wondering why the implementation looks so complicated. Wouldn't this be enough?
```
        long? ticksDifference = Subject?.Ticks - nearbyTime.Ticks;
        TimeSpan? difference = (ticksDifference != null) ? TimeSpan.FromTicks(Math.Abs(ticksDifference.Value)) : null;
```

Reply:
Hmm.. I like that approach :)

Reply:
This is to handle positive and negative overflow.
See #241

Reply:
I think this belongs to the difference calculation ~5 lines below? If not, I misundertand this completely 🙈

Reply:
I think I'm the one who got confused.

Reply:
😅

Reply:
... But I also got curious 😉 

and this [generates less code](https://sharplab.io/#v2:D4AQTAjAsAULDOAXATgVwMaIAQEFYG9YtisQBmLAFQEsBbAUwGUAHAQwDsB+LAIXoGEANgHt49SsIAUNBgHl2ggJ7dGqAEYArepgA0VOvXlKs7eq2RrFM+nussOWZsm3V41Yez0gIABixrtVlQxLABeLAAiCL02ZFZaeCxhTW1EAG0AXX9A4PocZABzeABKIhJCGBIqrBF2Au5EanQAa3gAEWoAM076Z3Z0ejCsVRTMTgA6GhbEgFoTMwsrA0mm1oBuMur9BnsuLAATLp6+gaHJRumO7t76fsGAQnD2VEFBYqxuOzZ2cYAxZGEtCmrUkAFlWIgABbjHBqeDnVbtI43O7jABqrEEqHoxXeAC4TC9BBtKlssJtqiAAOwHZEnegkqoAXwpFPI2yY324fCEonEwjA0gMRmUw3UWl0HJF83MlmstgMu0cznQrncnlIvmy6CCIXCURi5niiWSEvSWQCOty+SKpVJxAqZI5u24h2u9LOIzNWEehNe70+iu+fwBQMRYIh0Nh8K9qXRmOxK2mWDmpllSwYSdauKwBOer0ZZIpVWptPdtwGheILJgTKAA=)
```cs
TimeSpan? difference = (Subject != null) ? TimeSpan.FromTicks(Math.Abs(Subject.Value.Ticks - nearbyTime.Ticks)) : null;
```

Reply:
Ahh.. that's why I was handling this that way: The reason was, that the difference was always the longer differnence.

Like: Subject: 23:59:59, nearbyTime 00:01:00. The difference I want to see is 2 minutes and not 23:58:00 hours :)

Btw.. 23:58:00 hours difference is technically correct, but not that useful :)


Error with the old approach
`Expected time to be within 2m from <23:59:00.000>, but <00:01:00.000> was off by 2m.`
Error with the new approach
`Expected time to be within 2m from <23:59:00.000>, but <00:01:00.000> was off by 23h and 58m.`